test/e2e_clutz_dts_test.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+// tslint:disable:no-unused-expression mocha .to.be.empty getters.
+
+import {expect} from 'chai';
+import * as ts from 'typescript';
+
+import * as tsickle from '../src/tsickle';
+
+import {compileWithTransfromer, createProgram, dtsCompilerOptions} from './test_support';
+
+describe('clutz dts', () => {
```

Comment:
can you add a test that ensures the `.d.ts` files we emit are valid TS, and ideally one that shows they can be used?

Previously we've usually created a new test folder suffix (`.untyped`) to test variants of the emit, and then use golden file based tests. Maybe that would be a better approach here, too? It makes it easier to make an e2e with something using the `.d.ts` emit, and it keeps our tests consistently in one style. You'd need to change the test runner slight to support the pattern, but it might end up being not much more work than this here.

I'd also prefer if we had fewer one off test suites, and more code would use the golden file tests.

Reply:
I've moved all the tests that assert that strings appear in the .d.ts to be golden tests with the`.declaration` suffix.  I've also changed e2e_clutz_dts_test.ts to act similar to the previous `produces a valid .d.ts` test case, by checking for preemit errors after making a program with all the `.declaration` goldens.